### DISCUSSION TITLE
Dashboard on web site cannot initiate a game

### DISCUSSION THREAD

**Discussion by danconde** [azadbolour/eagerwords-community#8] [Q&A] [answered]
I somehow got into https://eagerwords.com/dashboard and noticed that I cannot invite a bot to play (button does nothing). I cannot also resume a game if I found one. I can only get a game by removing the /dashboard part of the URL and starting a game.

**Comment by azadbolour** [comment, marked as answer]
One guess is that you may have used the browser back button. EagerWords
does not support browser buttons and the user should be made aware of this
fact. Also, several sub-urls like https://eagerwords.com/dashboard are used
for the purpose of routing in the underlying React application, and are not
meant to be directly navigable by the user.

Azad

**Reply by danconde** [reply]
Thank you for reply.   I used a different browser instance in an incognito mode, and it entered the game. When I used the existing browser instance, the top level URL automatically redirected me to https://eagerwords.com/dashboard  I think I can play using incognito mode, or maybe clear cookies.

**Comment by azadbolour** [comment]
One thing to be aware of is that EagerWords is intended to be used from a
single browser instance. Its behavior with multiple simultaneous instances
is undefined.

**Reply by danconde** [reply]
I am *pretty sure* I was only playing one game (in one tab) at a time and I have reviewed my tabs to make sure there is only one.  Incidentally the dashboard sub-url shows the content as shown here.

![Screen Shot 2023-03-21 at 6 51 24 PM](https://user-images.githubusercontent.com/4314771/226781518-5830a428-473d-48de-8593-370a0fd1f2ab.png)

**Comment by azadbolour** [comment]
Looks like the bot ratings have been reset in one of the recent
deployments. I will initialize them in the next deployment - likely in
April.

Just FYI. The initial values are determined by having the bots at different
levels play rated games against each other.

Thanks for reporting it.

Azad

**Reply by danconde** [reply]
Browser based play seems to have something wrong, perhaps related to original error reported.
1. A laptop computer with connectivity displays when using https://eagerwords.com/play which I entered since the root URL keeps redirecting me to the dashboard.
![Screen Shot 2023-03-22 at 7 26 25 AM](https://user-images.githubusercontent.com/4314771/226948425-b45934d0-f86a-4b85-8d08-d6c62fe12b1c.png)

2. after pressing "signing in", got this error that is vague.
![Screen Shot 2023-03-22 at 7 05 16 AM](https://user-images.githubusercontent.com/4314771/226947925-4dc9b58f-ca9e-4945-890d-1e22db1e46e1.png)
and 
3 a guest game, which where one is not logged displays the following. It is not clear if a guest login expires?

![Screen Shot 2023-03-22 at 7 05 08 AM](https://user-images.githubusercontent.com/4314771/226948162-8cdec25b-4360-42d6-8d31-df127988070a.png)

**Reply by danconde** [reply]
> Looks like the bot ratings have been reset in one of the recent deployments. I will initialize them in the next deployment - likely in April. Just FYI. The initial values are determined by having the bots at different levels play rated games against each other. Thanks for reporting it. Azad
> […](#)
> On Tue, Mar 21, 2023 at 6:54 PM danconde ***@***.***> wrote: I am *pretty sure* I was only playing one game (in one tab) at a time and I have reviewed my tabs to make sure there is only one. Incidentally the dashboard sub-url shows the content as shown here. [image: Screen Shot 2023-03-21 at 6 51 24 PM] <https://user-images.githubusercontent.com/4314771/226781518-5830a428-473d-48de-8593-370a0fd1f2ab.png> — Reply to this email directly, view it on GitHub <[#8 (reply in thread)](https://github.com/azadbolour/eagerwords-community/discussions/8#discussioncomment-5387693)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAOFLY44QN7NQXZBYFRXASDW5JLWTANCNFSM6AAAAAAWB5CLWI> . You are receiving this because you commented.Message ID: <azadbolour/eagerwords-community/repo-discussions/8/comments/5387693@ github.com>

I don't know if you had re-initialized the bot ratings, but I saw all zeros again in your test web game.

![Screen Shot 2023-04-24 at 6 31 23 PM](https://user-images.githubusercontent.com/4314771/234164687-07a42cfa-3a06-48ba-9f5a-7abba24e0a86.png)

**Comment by azadbolour** [comment]
Thanks for reporting the issues. See below.

1. This can happen from time to time. The browser app does not attempt to
reconnect. It just gives up on a network error. This is a known issue. But
it seems to happen rarely in the browser. Hence has not been given high
priority yet.

2. This is a bug. But unfortunately there is not enough information to
pinpoint and fix it. In general, when something that looks like a bug
happens, if you could include the browser's console with the bug report,
that would help in troubleshooting the issue.

3. Under the covers the guest is also logged in and its login can expire.
The message should make this fact clear. Added to the backlog.

**Reply by danconde** [reply]
[eagerwords.com-1679507672120.log](https://github.com/azadbolour/eagerwords-community/files/11043255/eagerwords.com-1679507672120.log)
[eagerwords.com-1679507656052.log](https://github.com/azadbolour/eagerwords-community/files/11043257/eagerwords.com-1679507656052.log)

for item 2.

**Reply by danconde** [reply]
re: For the browser application, a copy of the logs reported in your browser console, your operating system and browser, and their versions.

macOS 11.17.4
Chrome 111.0.5563.64 (Official Build) (x86_64)

**Comment by azadbolour** [comment]
Thanks for the logs. They indicate an attempt to access a null reference. I
have added it to my todo list to diagnose and fix the issue.

Azad

**Reply by azadbolour** [reply]
On further investigation, the bug is what you had originally reported: somehow getting into the dashboard but not being able to do anything. I have traced this problem to the code for exiting a guest game and implemented the fix, which is now being tested, and is scheduled for release in May 2023.

The related problem you reported leading to error alerts stems from the direct use of "routing" URLs such as https://eagerwords.com/play. Those URLs are not meant to be used directly by the user in EagerWords. They are used in routing to different pages by the application itself, and their use is predicated on prerequisites that only the application can provide. Unfortunately, I know of no easy way to hide the routing URLs from the user. So the best we can do at this time is to document that they are not to be used directly.

Azad

**Comment by azadbolour** [comment]
Should be fixed now on the beta test site. Slated for release May 2023.
